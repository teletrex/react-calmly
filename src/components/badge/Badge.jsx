/* <LICENSE>
*
* Copyright (C) 2022 Louis F. Roehrs, All rights reserved.
*
* </LICENSE>
*  */




import React, { useCallback, useMemo } from 'react';
import PropTypes from 'prop-types';
import  settings  from '../../settings';
import classNames from 'classnames';

import { useTranslation } from '../../translation';
import { GRID_BREAKPOINTS } from '../constants';

import * as StatusIcons from './BadgeIcon';
import { VALID_STATUS, STATUS_ICON_POSITION } from './constants';

const { prefix } = settings;

function capitalizeFirstLetter(srcString) {
  return srcString.charAt(0).toUpperCase() + srcString.slice(1);
}

export const BadgePropTypes = {
  label: PropTypes.node,
  showLabel: PropTypes.bool,
  size: PropTypes.oneOf(['xs', 's', 'm', 'l']),
  status: PropTypes.oneOf(VALID_STATUS),
  statusIconPosition: PropTypes.oneOf([STATUS_ICON_POSITION.LEFT, STATUS_ICON_POSITION.RIGHT]),
  textOnly: PropTypes.bool,
  visibilityBreakpoint: PropTypes.oneOf(Object.values(GRID_BREAKPOINTS)),
};

export const BadgeDefaults = {
  label: '',
  showLabel: true,
  size: 's',
  status: 'published',
  statusIconPosition: STATUS_ICON_POSITION.LEFT,
  textOnly: false,
  visibilityBreakpoint: null,
};

const Badge = ({
  status,
  size,
  showLabel,
  label,
  className,
  textOnly,
  visibilityBreakpoint,
  statusIconPosition,
  ...otherProps
}) => {
  const { t } = useTranslation();

  const i18nStatuses = useMemo(
    () => ({
      disabled: t('disabled'),
      published: t('published'),
      approved: t('approved'),
      'draft , in Review': t('draft , in Review'),
      draft: t('draft'),
      pending: t('pending'),
      success: t('success'),
      cancelled: t('cancelled'),
      retired: t('retired'),
      scheduled: t('scheduled'),
      'draft / Open': t('draft / Open'),
      'inprogress / Sending': t('inprogress / Sending'),
      failed: t('failed'),
      complete: t('complete'),
      enabled: t('enabled'),
      routine: t('routine'),
      event: t('event'),
      paused: t('paused'),
    }),
    [t]
  );

  const getStatusIcon = useCallback(() => {
    if (!Object.keys(i18nStatuses).includes(status)) return null;

    const MyStatusIcon =
      StatusIcons[capitalizeFirstLetter(status.replace(/ .*/, '')) + size.toUpperCase()];
    return (
      <MyStatusIcon
        className={classNames(`${prefix}--status__icon`, `${prefix}--status__${size}icon`, {
          [`${prefix}--status__${size}icon-right`]:
            statusIconPosition === STATUS_ICON_POSITION.RIGHT,
        })}
      />
    );
  }, [i18nStatuses, size, status, statusIconPosition]);

  const actualLabel = useMemo(
    () => (showLabel && !label ? i18nStatuses[status].replace(/^\w/, c => c.toUpperCase()) : label),
    [i18nStatuses, label, showLabel, status]
  );

  return showLabel ? (
    <div
      {...otherProps}
      className={classNames(`${prefix}--status`, className, {
        [`${prefix}--status__icon-right`]: statusIconPosition === STATUS_ICON_POSITION.RIGHT,
        [`${prefix}--status__hide-${visibilityBreakpoint}`]: visibilityBreakpoint,
      })}
    >
      {!textOnly && statusIconPosition === STATUS_ICON_POSITION.LEFT && getStatusIcon()}
      <div className={`${prefix}--status__${size}text`}>
        <p className={`${prefix}--status__${size}text--content`}>{actualLabel}</p>
      </div>
      {!textOnly && statusIconPosition === STATUS_ICON_POSITION.RIGHT && getStatusIcon()}
    </div>
  ) : (
    !textOnly && getStatusIcon()
  );
};

Badge.propTypes = {
  ...BadgePropTypes,
};

Badge.defaultProps = {
  ...BadgeDefaults,
};

export default Badge;
