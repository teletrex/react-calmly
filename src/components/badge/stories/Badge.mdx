import { Canvas, Story } from '@storybook/addon-docs/blocks';

# Badge



Badges are generally used to visually identify status of a process or object, for example, a report readiness status, data job status, an email status or a webpage status.
Status icons can be paired with the text or stand alone.

## Default

By default `Badge` renders as `published` with `label` visible.

<Canvas>
  <Story id="atoms-badge--default" />
</Canvas>


## Component API

| Parameter name     |             Type             |  Default value  | Required  |                                  Description                                      |
| -------------------| :--------------------------: | :-------------: | :-------: | --------------------------------------------------------------------------------- |
| label              | *OneOf(string, node)*        | `''`            | No        | Badge label                                                                       |
| showLabel          | *boolean*                    | `true`          | No        | Show label flag                                                                   |
| size               | *OneOf('xs', 's', 'm', 'l')* | `s`             | No        | Size of Badge                                                                     |
| status             | *OneOf(STATUS_TYPE)*         | `'published'`   | No        | Status of the Badge (list of predefined labels). Available statuses listed below. |
| statusIconPosition | *OneOfType('left', 'right')* | `left`          | No        | Position of the status icon (on the left or right side of the label)              |
| textOnly           | *boolean*                    | `false`         | No        | Show status icon flag                                                             |

### `status` (STATUS_TYPE)

List of available statuses:
- `disabled`
- `published`
- `approved`
- `draft , in Review`
- `draft`
- `pending`
- `success`
- `cancelled`
- `scheduled`
- `draft / Open`
- `inprogress / Sending`
- `failed`
- `complete`
- `enabled`
- `routine`
- `event`
- `retired`
- `paused`


