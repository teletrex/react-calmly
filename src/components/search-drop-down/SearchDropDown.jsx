/* <LICENSE>
*
* Copyright (C) 2022 Louis F. Roehrs, All rights reserved.
*
* </LICENSE>
*  */



import React, { useCallback, useEffect, useMemo, useRef, useState } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import  settings  from '../../settings';
import noop from 'lodash/noop';
import isObjectLike from 'lodash/isObjectLike';
import { keys, match, matches } from '../keyboard';
import uniqueId from 'lodash/uniqueId';

import { useTranslation } from '../../translation';
import {Search} from '@carbon/react';
import { useDropdownInPortal, WithPortalPropType } from '../../utils/hooks/use-dropdown-in-portal';
import { DomElement } from '../../utils/DomUtils';
import { usePrevious } from '../../utils/hooks';

const { prefix } = settings;
const mainPrefix = `${prefix}--search-drop-down`;

const SIZE = {
  SMALL: 'sm',
  LARGE: 'lg',
};

const KIND = {
  PRIMARY: 'primary',
  SECONDARY: 'secondary',
};

const CONTROL_KEYS = [keys.ArrowUp, keys.ArrowDown, keys.Enter, keys.Escape];

const SearchDropDown = ({
  className,
  disabled,
  dropdownMaxHeight,
  children,
  id,
  kind,
  labelText,
  onBlur,
  onChange,
  onFocus,
  onKeyDown,
  selectedElementCallback,
  size,
  withBorderBottom,
  shouldShowDropdown,
  showDropdownOnInputFocus,
  handleSelect,
  withPortal,
  onDropDownOpenChange,
  ...rest
}) => {
  const { t } = useTranslation();
  const root = useRef();
  const [input, setInput] = useState('');

  const [focusIndex, setFocusIndex] = useState(-1);
  const iterableElements = [];

  const onDropDownOpenChangeHandler = useRef(onDropDownOpenChange);
  onDropDownOpenChangeHandler.current = onDropDownOpenChange;
//  TODO: really should just reuse a menu or list box to get the styles right.

  const {
    Portal,
    portalRef,
    showDropDown,
    isDropDownOpen,
    dropDownInPortalOffset,
  } = useDropdownInPortal(root, withPortal, {
    closeIfClickOutside: true,
  });

  const isDropDownWasOpen = usePrevious(isDropDownOpen, isDropDownOpen);

  const usePortalDropDown = useMemo(() => {
    return withPortal === true || isObjectLike(withPortal);
  }, [withPortal]);

  const labelTextOrDefault = useMemo(() => {
    return labelText || t('search');
  }, [labelText, t]);

  const calculateNextIndex = (currentIndex, maxIndex, isDown) => {
    if (isDown) {
      if (currentIndex + 1 >= maxIndex) {
        return 0;
      }
      return currentIndex + 1;
    }
    if (currentIndex - 1 < 0) {
      return maxIndex - 1;
    }
    return currentIndex - 1;
  };

  const handleKeyDown = useCallback(
    e => {
      onKeyDown(e);

      if (matches(e, CONTROL_KEYS)) {
        const isArrowUp = match(e.which, keys.ArrowUp);
        const isArrowDown = match(e.which, keys.ArrowDown);
        const isEnter = match(e.which, keys.Enter);
        const isEscape = match(e.which, keys.Escape);

        if (isArrowUp || isArrowDown) {
          setFocusIndex(calculateNextIndex(focusIndex, iterableElements.length, isArrowDown));
        } else if (isEscape) {
          showDropDown(false);
        } else if (isEnter) {
          if (focusIndex === -1) {
            handleSelect(input);
          } else {
            selectedElementCallback({
              i: focusIndex,
              element: iterableElements[focusIndex],
            });
          }
          DomElement(document.activeElement).blur();
        }
      }
    },
    [
      focusIndex,
      handleSelect,
      input,
      iterableElements,
      onKeyDown,
      selectedElementCallback,
      showDropDown,
    ]
  );

  const searchDropDownClasses = useMemo(
    () =>
      classNames(className, {
        [`${mainPrefix}--small`]: size === SIZE.SMALL,
        [`${mainPrefix}--large`]: size === SIZE.LARGE,
        [`${mainPrefix}--primary`]: kind === KIND.PRIMARY,
        [`${mainPrefix}--secondary`]: kind === KIND.SECONDARY,
      }),
    [className, kind, size]
  );

  const dropdownStyle = useMemo(() => (dropdownMaxHeight ? { maxHeight: dropdownMaxHeight } : {}), [
    dropdownMaxHeight,
  ]);

  const isDropDownOpenRequire = useCallback(
    (triggerEvent, inputString = input) => {
      return (
        shouldShowDropdown({ ...rest, showDropdownOnInputFocus, triggerEvent, inputString }) ===
        true
      );
    },
    [input, rest, shouldShowDropdown, showDropdownOnInputFocus]
  );

  const listMouseDown = useCallback(e => {
    e.preventDefault();
  }, []);

  const handleInputBlur = useCallback(
    e => {
      showDropDown(false);
      setFocusIndex(-1);
      onBlur(e);
    },
    [onBlur, showDropDown]
  );

  const handleInputFocus = useCallback(
    e => {
      if (isDropDownOpenRequire(e)) {
        showDropDown(true);
      }
      onFocus(e);
    },
    [isDropDownOpenRequire, onFocus, showDropDown]
  );

  const handleInputChange = useCallback(
    e => {
      if (e && e.target) {
        const inputValue = e.target.value;
        setInput(inputValue);
        showDropDown(isDropDownOpenRequire(e, inputValue));
      }
      onChange(e);
    },
    [showDropDown, isDropDownOpenRequire, onChange]
  );

  useEffect(() => {
    if (isDropDownOpen !== isDropDownWasOpen) {
      onDropDownOpenChangeHandler.current(isDropDownOpen);
    }
  }, [isDropDownOpen, isDropDownWasOpen]);

  const childrenClone =
    React.Children.count(children) > 0
      ? React.Children.map(children, group => {
          return React.isValidElement(group)
            ? React.cloneElement(group, {
                ...group.props,
                children: React.Children.map(group.props.children, child => {
                  if (child && child.props.clickable) {
                    iterableElements.push(child);
                  }
                  const index = iterableElements.length - 1;
                  return React.isValidElement(child)
                    ? React.cloneElement(child, {
                        ...child.props,
                        selected: child.props.clickable && focusIndex === index,
                        onMouseDown: child.props.clickable
                          ? () => {
                              showDropDown(false);
                              setFocusIndex(-1);
                              selectedElementCallback({
                                i: index,
                                element: child,
                              });
                            }
                          : undefined,
                      })
                    : null;
                }),
              })
            : null;
        })
      : null;

  return (
    <>
      {labelText && (
        <label className={`${prefix}--label`} htmlFor={`search-${id}`} id={`${id}-label`}>
          {t(labelText)}
        </label>
      )}
      <div
        ref={root}
        className={classNames(mainPrefix, searchDropDownClasses, {
          [`${mainPrefix}--underlined`]: withBorderBottom,
        })}
      >
        <Search
          aria-label={labelTextOrDefault}
          {...rest}
          disabled={disabled}
          handleSelect={handleSelect}
          id={`search-${id}`}
          labelText=""
          light={kind === KIND.SECONDARY}
          onBlur={handleInputBlur}
          onChange={handleInputChange}
          onFocus={handleInputFocus}
          onKeyDown={handleKeyDown}
          small={size === SIZE.SMALL}
        />
        {usePortalDropDown ? (
          <Portal>
            {isDropDownOpen && (
              <div
                ref={portalRef}
                className={classNames(
                  mainPrefix,
                  searchDropDownClasses,
                  `${mainPrefix}--in-portal`
                )}
                style={dropDownInPortalOffset}
              >
                <div
                  className={`${mainPrefix}--list`}
                  onMouseDown={listMouseDown}
                  role="presentation"
                  style={dropdownStyle}
                >
                  {isDropDownOpen && childrenClone}
                </div>
              </div>
            )}
          </Portal>
        ) : (
          <div
            className={`${mainPrefix}--list`}
            onMouseDown={listMouseDown}
            role="presentation"
            style={dropdownStyle}
          >
            {isDropDownOpen && childrenClone}
          </div>
        )}
      </div>
    </>
  );
};

SearchDropDown.shouldShowDropdownOnFocusDefault = ({ showDropdownOnInputFocus, triggerEvent }) => {
  return triggerEvent.type === 'focus' ? showDropdownOnInputFocus : false;
};

SearchDropDown.shouldShowDropdownOnChangeDefault = ({ triggerEvent }) => {
  return triggerEvent.type === 'change';
};

SearchDropDown.shouldShowDropdownDefault = params =>
  SearchDropDown.shouldShowDropdownOnFocusDefault(params) ||
  SearchDropDown.shouldShowDropdownOnChangeDefault(params);

SearchDropDown.propTypes = {
  children: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node]),
  className: PropTypes.string,
  disabled: PropTypes.bool,
  // eslint-disable-next-line react/require-default-props
  dropdownMaxHeight: PropTypes.string,
  handleSelect: PropTypes.func,
  id: PropTypes.string,
  kind: PropTypes.oneOf([KIND.PRIMARY, KIND.SECONDARY]),
  // eslint-disable-next-line react/require-default-props
  labelText: PropTypes.string,
  onBlur: PropTypes.func,
  onChange: PropTypes.func,
  onDropDownOpenChange: PropTypes.func,
  onFocus: PropTypes.func,
  onKeyDown: PropTypes.func,
  selectedElementCallback: PropTypes.func,
  shouldShowDropdown: PropTypes.func,
  showDropdownOnInputFocus: PropTypes.bool,
  size: PropTypes.oneOf([SIZE.SMALL, SIZE.LARGE]),
  withBorderBottom: PropTypes.bool,
  withMagnifyingGlass: PropTypes.bool,
  withPortal: WithPortalPropType,
};

SearchDropDown.defaultProps = {
  children: null,
  className: '',
  disabled: false,
  handleSelect: noop,
  id: uniqueId('search__drop_down__id_'),
  kind: KIND.PRIMARY,
  onBlur: noop,
  onChange: noop,
  onDropDownOpenChange: noop,
  onFocus: noop,
  onKeyDown: noop,
  selectedElementCallback: noop,
  shouldShowDropdown: SearchDropDown.shouldShowDropdownDefault,
  showDropdownOnInputFocus: false,
  size: SIZE.LARGE,
  withBorderBottom: false,
  withMagnifyingGlass: true,
  withPortal: false,
};

export const SearchDropDownPortalAnchor = React.forwardRef(({ className }, ref) => {
  return (
    <div style={{ position: 'relative', display: 'contents' }}>
      <div
        ref={ref}
        className={classNames(`${mainPrefix}-portal-anchor`, { [className]: className })}
        style={{ position: 'absolute' }}
      />
    </div>
  );
});

export default SearchDropDown;
