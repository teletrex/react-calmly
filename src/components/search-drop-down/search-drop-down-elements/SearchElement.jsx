/* <LICENSE>
*
* Copyright (C) 2022 Louis F. Roehrs, All rights reserved.
*
* </LICENSE>
*  */



import React, { useEffect, useRef } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import  settings  from '../../../settings';
import noop from 'lodash/noop';

const { prefix } = settings;

const SearchElement = props => {
  const { children, onMouseDown, selected, onClick, clickable } = props;
  const element = useRef(null);

  const searchElementClasses = classNames(`${prefix}--search-element`, {
    [`${prefix}--search-element--selected`]: selected,
  });

  useEffect(() => {
    if (selected && element.current) {
      element.current.scrollIntoView(false);
    }
  }, [selected, clickable]);

  return (
    <div
      ref={element}
      className={searchElementClasses}
      onClick={onClick}
      onMouseDown={onMouseDown}
      onTouchStart={onMouseDown}
      role="presentation"
    >
      {children}
    </div>
  );
};

SearchElement.propTypes = {
  children: PropTypes.node.isRequired,
  // eslint-disable-next-line react/no-unused-prop-types
  clickable: PropTypes.bool,
  onClick: PropTypes.func,
  onMouseDown: PropTypes.func,
  selected: PropTypes.bool,
};

SearchElement.defaultProps = {
  clickable: false,
  onClick: noop,
  onMouseDown: noop,
  selected: false,
};

export default SearchElement;
