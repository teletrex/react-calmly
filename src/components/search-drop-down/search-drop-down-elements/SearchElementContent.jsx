/* <LICENSE>
*
* Copyright (C) 2022 Louis F. Roehrs, All rights reserved.
*
* </LICENSE>
*  */



import React from 'react';
import PropTypes from 'prop-types';
import  settings  from '../../../settings';

const { prefix } = settings;

const SearchElementContent = props => {
  const { children } = props;

  return <div className={`${prefix}--search-element__content`}>{children}</div>;
};

SearchElementContent.propTypes = {
  children: PropTypes.node.isRequired,
};

export default SearchElementContent;
