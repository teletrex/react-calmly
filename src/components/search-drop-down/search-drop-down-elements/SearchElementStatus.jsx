/* <LICENSE>
*
* Copyright (C) 2022 Louis F. Roehrs, All rights reserved.
*
* </LICENSE>
*  */



import React from 'react';
import PropTypes from 'prop-types';
import  settings  from '../../../settings';

const { prefix } = settings;

const SearchElementStatus = props => {
  const { children } = props;

  return <div className={`${prefix}--search-element__status`}>{children}</div>;
};

SearchElementStatus.propTypes = {
  children: PropTypes.node.isRequired,
};

export default SearchElementStatus;
