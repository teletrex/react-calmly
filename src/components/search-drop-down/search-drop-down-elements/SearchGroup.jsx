/* <LICENSE>
*
* Copyright (C) 2022 Louis F. Roehrs, All rights reserved.
*
* </LICENSE>
*  */



import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import  settings  from '../../../settings';

const { prefix } = settings;

const SearchGroup = props => {
  const { children, className } = props;

  return <div className={classNames(className, `${prefix}--search-group`)}>{children}</div>;
};

SearchGroup.propTypes = {
  children: PropTypes.node.isRequired,
  className: PropTypes.string,
};

SearchGroup.defaultProps = {
  className: '',
};

export default SearchGroup;
