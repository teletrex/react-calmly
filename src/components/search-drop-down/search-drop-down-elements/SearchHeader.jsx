/* <LICENSE>
*
* Copyright (C) 2022 Louis F. Roehrs, All rights reserved.
*
* </LICENSE>
*  */



import React from 'react';
import PropTypes from 'prop-types';
import  settings  from '../../../settings';

const { prefix } = settings;

const SearchHeader = props => {
  const { children } = props;

  return <div className={`${prefix}--search-header`}>{children}</div>;
};

SearchHeader.propTypes = {
  children: PropTypes.node.isRequired,
};

export default SearchHeader;
