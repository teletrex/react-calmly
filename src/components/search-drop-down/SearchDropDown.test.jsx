/* <LICENSE>
* 
* Copyright (C) 2022 Louis F. Roehrs, All rights reserved.
* 
* </LICENSE>
*  */



import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { keys } from 'carbon-components-react/es/internal/keyboard';

import Badge from '../badge';

import SearchDropDown, { SearchDropDownPortalAnchor } from './SearchDropDown';
import SearchHeader from './search-drop-down-elements/SearchHeader';
import SearchElement from './search-drop-down-elements/SearchElement';
import SearchElementContent from './search-drop-down-elements/SearchElementContent';
import SearchElementStatus from './search-drop-down-elements/SearchElementStatus';
import SearchGroup from './search-drop-down-elements/SearchGroup';

describe('SearchDropDown', () => {
  const props = {
    handleSelect: jest.fn(),
    selectedElementCallback: jest.fn(),
    showDropdownOnInputFocus: false,
  };

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('verify if keydown work', () => {
    const { getByLabelText } = render(
      <SearchDropDown {...props}>
        <SearchGroup>
          <SearchElement clickable>
            <SearchElementContent>Lorem Ipsum</SearchElementContent>
          </SearchElement>
        </SearchGroup>
      </SearchDropDown>
    );

    fireEvent.keyDown(getByLabelText('search'), keys.Enter);

    expect(props.handleSelect).toHaveBeenCalledTimes(1);
  });

  it('verify if keydown work on sub element', () => {
    const { getByLabelText, getByText } = render(
      <SearchDropDown {...props}>
        <SearchGroup>
          <SearchElement clickable>
            <SearchElementContent>Lorem Ipsum</SearchElementContent>
          </SearchElement>
          <SearchElement clickable>
            <SearchElementContent>Dolor sit amet</SearchElementContent>
          </SearchElement>
          <SearchElement clickable>
            <SearchElementContent>Consectetur adipiscing</SearchElementContent>
          </SearchElement>
        </SearchGroup>
      </SearchDropDown>
    );

    fireEvent.change(getByLabelText('search'), { target: { value: 'test' } });
    expect(getByText('Lorem Ipsum')).toBeInTheDocument();
    expect(getByText('Dolor sit amet')).toBeInTheDocument();
    expect(getByText('Consectetur adipiscing')).toBeInTheDocument();
  });

  it('verify if SearchDropDown match snapshot', () => {
    const { container } = render(
      <SearchDropDown id="search_drop-down-1">
        <SearchGroup>
          <SearchHeader>Recent</SearchHeader>
          <SearchElement clickable>
            <SearchElementContent>Lorem Ipsum</SearchElementContent>
            <SearchElementStatus>
              <Badge label="New" />
            </SearchElementStatus>
          </SearchElement>
          <SearchElement clickable>
            <SearchElementContent>Dolor sit amet</SearchElementContent>
            <SearchElementStatus>Removed</SearchElementStatus>
          </SearchElement>
          <SearchElement clickable>
            <SearchElementContent>Consectetur adipiscing</SearchElementContent>
          </SearchElement>
        </SearchGroup>
      </SearchDropDown>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('verify if SearchDropDownPortalAnchor match snapshot', () => {
    const { container } = render(<SearchDropDownPortalAnchor className="tst-class" />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
