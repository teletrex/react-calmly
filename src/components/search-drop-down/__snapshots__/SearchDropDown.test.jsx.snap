// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchDropDown verify if SearchDropDown match snapshot 1`] = `
<div
  class="bx--search-drop-down bx--search-drop-down--large bx--search-drop-down--primary"
>
  <div
    aria-label=""
    class="bx--search bx--search--xl"
    role="search"
  >
    <svg
      aria-hidden="true"
      aria-label=""
      class="bx--search-magnifier"
      focusable="false"
      height="16"
      preserveAspectRatio="xMidYMid meet"
      role="img"
      style="will-change: transform;"
      viewBox="0 0 16 16"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M15 14.3L10.7 10c1.9-2.3 1.6-5.8-.7-7.7S4.2.7 2.3 3 .7 8.8 3 10.7c2 1.7 5 1.7 7 0l4.3 4.3.7-.7zM2 6.5C2 4 4 2 6.5 2S11 4 11 6.5 9 11 6.5 11 2 9 2 6.5z"
      />
    </svg>
    
    <input
      aria-label="search"
      class="bx--search-input"
      i18n="[object Object]"
      id="search-search_drop-down-1"
      placeholder=""
      title=""
      type="text"
      value=""
    />
    <button
      aria-label="Clear search input"
      class="bx--search-close bx--search-close--hidden"
      title="Clear input"
      type="button"
    >
      <svg
        aria-label="Clear search input"
        focusable="false"
        height="20"
        preserveAspectRatio="xMidYMid meet"
        role="img"
        style="will-change: transform;"
        viewBox="0 0 32 32"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M24 9.4L22.6 8 16 14.6 9.4 8 8 9.4l6.6 6.6L8 22.6 9.4 24l6.6-6.6 6.6 6.6 1.4-1.4-6.6-6.6L24 9.4z"
        />
      </svg>
    </button>
  </div>
  <div
    class="bx--search-drop-down--list"
    role="presentation"
  />
</div>
`;

exports[`SearchDropDown verify if SearchDropDownPortalAnchor match snapshot 1`] = `
<div
  style="position: relative; display: contents;"
>
  <div
    class="bx--search-drop-down-portal-anchor tst-class"
    style="position: absolute;"
  />
</div>
`;
